modemmanager: 1.7.990 -> 1.10.0

worldofpeace opened the pull request:
###### Motivation for this change
Fix https://github.com/NixOS/nixpkgs/issues/57615

I still need to give the News items a better look.

Probably should add multiple outputs for `modemmanager`?

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS (master)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

worldofpeace committed:
libmbim: 1.16.2 -> 1.18.0

worldofpeace committed:
libqmi: 1.20.2 -> 1.22.2

Drop --enable-mbim-qmux because we have libmbim >= 1.14.0 [0]

[0]: https://gitlab.freedesktop.org/mobile-broadband/libqmi/blob/1.16.0/NEWS#L5

jtojnar commented on pkgs/development/libraries/libqmi/default.nix:
```diff
@@ -1,23 +1,36 @@
 
-  preBuild = ''
+  preConfigure = ''
     patchShebangs .
```
This should be more specific to avoid overpatching (not sure if this is actually a problem) and make reasoning about this cleaner. Also, this is usually done in `postPatch`.

worldofpeace replied:
Only files I see that might need this are
* https://gitlab.freedesktop.org/mobile-broadband/libmbim/blob/master/build-aux/mbim-codegen/mbim-codegen
* https://gitlab.freedesktop.org/mobile-broadband/libqmi/blob/master/build-aux/qmi-codegen/qmi-codegen
* https://gitlab.freedesktop.org/mobile-broadband/libqmi/blob/master/utils/qmidb/qmidb.py

Tried dropping it entirely and didn't have an issue building.

jtojnar commented on pkgs/tools/networking/modem-manager/default.nix:
```diff
@@ -1,38 +1,37 @@
   ];
 
-  installFlags = [ "DESTDIR=\${out}" ];
+  installFlags = [ "DESTDIR=${placeholder ''out''}" ];
```
We would want to avoid this as described in https://github.com/NixOS/nixpkgs/pull/51382#issuecomment-443504896 (even if we keep the wrong service name for now).

jtojnar commented:
The rest looks fine after consulting the news:

* https://gitlab.freedesktop.org/mobile-broadband/libmbim/blob/1.18.0/NEWS
* https://gitlab.freedesktop.org/mobile-broadband/libqmi/blob/1.22.2/NEWS
* https://gitlab.freedesktop.org/mobile-broadband/ModemManager/blob/1.10.0/NEWS

worldofpeace commented:
Think I did all the things, just need to verify that we actually fixed https://github.com/NixOS/nixpkgs/issues/41055 by linking the service relatively.

jtojnar commented on pkgs/tools/networking/modem-manager/default.nix:
```diff
@@ -1,56 +1,48 @@
+  doCheck = true;
 
   preInstall = ''
     mkdir -p $out/etc/systemd/system
```
Do we still need this?

worldofpeace replied:
No, dropped it

worldofpeace commented:
Still see the issue described at https://github.com/NixOS/nixpkgs/issues/41055#issuecomment-394640702 so I'm not going to link it relatively.

worldofpeace committed:
modemmanager: 1.7.990 -> 1.10.0

* Add flags for systemd [0]

* Enable tests
  Though I actually didn't look to see how they're
  performed, only effort done was to get a success-
  ful build.

* Enable vala bindings

[0]: https://gitlab.freedesktop.org/mobile-broadband/ModemManager/blob/1.8.0/NEWS#L21

jtojnar commented on pkgs/development/libraries/libqmi/default.nix:
```diff
@@ -1,23 +1,32 @@
+  doCheck = true;
 
   meta = with stdenv.lib; {
     homepage = http://www.freedesktop.org/wiki/Software/libqmi/;
```
```suggestion
    homepage = https://www.freedesktop.org/wiki/Software/libqmi/;
```

jtojnar commented on pkgs/development/libraries/libmbim/default.nix:
```diff
@@ -1,21 +1,32 @@
+  doCheck = true;
 
   meta = with stdenv.lib; {
     homepage = http://www.freedesktop.org/software/libmbim/;
```
```suggestion
    homepage = https://www.freedesktop.org/wiki/Software/libmbim/;
```

jtojnar reviewed (approved):
Thank you very much for taking care of this.

worldofpeace commented:
Thanks. Feel free to open more issues for things you'd like taken care of.

flokli commented:
@jtojnar should/can this be backported to 19.03 as well?

jtojnar commented:
Probably should.

flokli commented:
Backported to 19.03 in 2b1ee54b873e3e5777ed92cdefbffa86f044ffe2, 4cec8a5fcb9b2802c3cad03101b221778994c091 and 65028ce1c91613dab293b1a9cf7a02bc97b3956c.

Schweber commented:
I'm sorry, what is the solution now? How do i disable modem manager?

jtojnar commented:
@Schweber you are probably looking for #214086

Schweber commented:
@jtojnar I'm looking for a way to **uninstall** modemmanager if possible. Is there an option for `configuration.nix` (i didn't find one)?

jtojnar commented:
@Schweber lot of software depends on it so there is no easy way. And for questions, it is better to open a thread on [Discourse](https://discourse.nixos.org/) rather than bumping a four year old unrelated pull request.
